### PR TITLE
Add Baystate Health

### PR DESCRIPTION
**Pull request by kriation**
This is an initial PR for #11. The scraper returns a boolean for availability based on the Registration banner in the sign-up page.
I'll continue working on returning more detail once appointments are made available.

**Output of test run**

**Screenshot of site during test run**
Baystate Health-2021-02-27T215940Z

### CODE REVIEW

File: data/sites.json [hunk 1]
```diff
@@ -305,5 +305,9 @@
             { "street": "166 Walnut St", "city": "Saugus", "zip": "01906" },
             { "street": "296 Buffinton St", "city": "Somerset", "zip": "02726" }
         ]
+    },
+    "BaystateHealth": {
+        "name": "Baystate Health",
+        "website": "https://www.baystatehealth.org/covid19"
```
Comment (livgust) [inline data/sites.json:311]: I think you want this to be signUpLink - we should always return a signUpLink that will be the website users are kicked out to when appointments are available and they try to book.

File: site-scrapers/BaystateHealth.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    return {
+        ...sites.BaystateHealth,
+        ...webData,
+        timestamp: new Date(),
+    };
+};
+
+async function ScrapeWebsiteData(browser) {
+    const page = await browser.newPage();
+    const signUpPageSelector = "div > div > section > div > div > p > a";
```
Comment (livgust) [inline site-scrapers/BaystateHealth.js:19]: did you notice anything more descriptive we could use to define this? like an ID or class name that is a little more specific and less prone to error/duplication?

**Review by livgust — CHANGES_REQUESTED** [review]
Hi @kriation , thanks for this! Please review my comments and let me know if you have any questions.
In addition, we changed the format of the scraper files to be like this:
site-scrapers/BaystateHealth/[index|config].js. If you pull in master you'll see.

File: site-scrapers/BaystateHealth.js (hunk 2 above)
Comment (kriation) [reply site-scrapers/BaystateHealth.js:19]: The parent divs and section do have class names that I can use. I'll incorporate them.

File: site-scrapers/BaystateHealth/index.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            ...site,
+            ...webData,
+            timestamp: new Date(),
+        };
+    }
+};
+
+async function ScrapeWebsiteData(browser) {
+    let pageArray = await browser.pages();
+    const page = pageArray[1];
```
Comment (livgust) [inline site-scrapers/BaystateHealth/index.js:36]: I don't think we can count on this. When the scrapers run they open up a LOT of pages. You should just move lines 22 and 23 into ScrapeWebsiteData, or pass the page through to that function.

File: site-scrapers/BaystateHealth/index.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        "app-guest-covid-vaccine-register.ion-page > div.sky-bg > div.content-body > div.content-card > h3.text-align-center"
+    );
+    let alert = alertElement
+        ? await alertElement.evaluate((node) => node.innerText)
+        : false;
+    let hasAppointments = false;
+    let totalAvailability = 0;
+    let date = false;
+    if (!alert) {
+        // TBD: Are there appointments?
```
Comment (livgust) [inline site-scrapers/BaystateHealth/index.js:48]: Have you considered saving a screenshot or HTML data to S3 or posting a notice to Slack? See util/s3.js and util/slack.js for ideas. Tagging @zpeyton who created those utilities

File: site-scrapers/BaystateHealth/index.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+async function ScrapeWebsiteData(browser) {
+    let pageArray = await browser.pages();
+    const page = pageArray[1];
+    // Is the clinic open?
+    const alertElement = await page.$(
+        "app-guest-covid-vaccine-register.ion-page > div.sky-bg > div.content-body > div.content-card > h3.text-align-center"
+    );
+    let alert = alertElement
```
Comment (livgust) [inline site-scrapers/BaystateHealth/index.js:41]: this can be a const

File: site-scrapers/BaystateHealth/index.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        ? await alertElement.evaluate((node) => node.innerText)
+        : false;
+    let hasAppointments = false;
+    let totalAvailability = 0;
+    let date = false;
```
Comment (livgust) [inline site-scrapers/BaystateHealth/index.js:46]: could we remove this unused variable?

**Review by livgust — CHANGES_REQUESTED** [review]
Hi @kriation , thanks for this! I have a few nitpicky comments now with your latest version of this scraper.

**Comment by kriation** [comment]
Malfunctioning test for Heywood scraper is resolved in #147

**Review by livgust — APPROVED** [review]

**Pull request merged**